apps/extension/src/core/managers/base.ts

```diff
@@ -77,7 +79,7 @@ export abstract class BaseManager<T extends BaseModel> {
     key: string | null,
     indexName: string
   ): Promise<boolean> {
-    const index = key ? `${indexName}-${key}` : indexName
+    const index = this._getIndex(indexName, key || undefined)
```

Comment:
Empty string, undefined or null are all falsey in js, so prob don't need the fallback to undefined.

Reply:
needed this due to null vs undefined types, but decided to change `null` to be `undefined` in this method's signature to make it less confusing